index.html

```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
+                            Solid WebID Profiles and share the same data model. <a href="reader-application">Reader
+                                Applications</a> retrieve representations of Extended Profile Documents the same way as they
+                            retrieve Solid WebID Profiles.</p>
+                    </div>
+                    <p about="" id="reader-application-extended-profile" rel="spec:requirement"
+                        resource="reader-application-extended-profile">
+                        <span property="spec:statement"><a rel="spec:requirementSubject" href="#ReaderApplication">Reader
+                                Application</a> <span rel="spec:requirementLevel" resource="spec:MAY">MAY</span> retrieve a
+                            representation of an <a href="#extended-profile-documents">Extended Profile Document</a>.</span>
+                    </p>
+                </section>
+                <section id="updating-profile" inlist="" rel="schema:hasPart" resource="#updating-profile">
+                    <h3 property="schema:name">Updating Profile</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p><a href="#writer-applications">Writer Applications</a> perform <a href="https://solidproject.org/TR/protocol#writing-resources">write operations</a> against Solid Protocol <a href="https://solidproject.org/TR/protocol#Server">servers</a>.</p>
+                        <div class="note" id="prioritize-webid-profile" inlist="" rel="schema:hasPart"
+                            resource="#prioritize-webid-profile">
+                            <h5 property="schema:name"><span>Note</span>:</h5>
+                            <div datatype="rdf:HTML" property="schema:description">
+                                <p>To promote self-describing resources and efficient discovery and reuse of profile
+                                    information, implementations and authors are encouraged to prioritize using the <a
+                                        href="#solid-profile">Solid WebID Profile</a> before needing to use the <a
+                                        href="extended-profile-documents">Extended WebID Profile.</a></p>
+                            </div>
+                        </div>
+                        </div>
+                        <p about="" id="writer-application-webid-profile" rel="spec:requirement"
+                            resource="writer-application-webid-profile">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
+                                an <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be
+                                updated.</span>
```

Comment:
> Writer Application MUST use an HTTP PUT or PATCH request targeting the Solid WebID Profile to be updated

Ideally requirement statements should be standalone statements to make them easier to understand in other contexts such as test results. What I think is missing above is the clear purpose - the context is 'update' but perhaps it could be worded as "To update the Solid WebID Profile ...". Is there also something about the body and content type of such requests.

Reply:
Writer Application is a Solid app (Solid Protocol). But we need to revisit this with respect to the Solid Protocol because it is not currently defined as an independent product class. Writer Application could be a specialisation of Solid Protocol Client but then that would still require us to revisit Solid app in Solid Protocol. The general understanding is that the Solid app issues HTTP requests through the Solid Protocol Client against a Solid Protocol Server. So, Writer Application writes to a Solid WebID Profile on a Solid storage.

Solid WebID Profile is an RDF document. Solid Servers can potentially accept any concrete RDF syntax, but only required to respond to Turtle and JSON-LD. The body of the request includes an RDF representation based on the Solid WebID Profile data model.

---

I agree with the suggestion to have multiple (identifiable) requirements. And doing that for all similar requirements. (It is also okay to leave it as is for the time being - not super urgent per se.)

---

Purpose at the front instead of at the end seems okay too. I would only suggest that the specification should generally be consistent (and boring), so if "purpose" comes first, do it for all similar requirements, e.g., "[purpose] [requirementSubject] [requirementLevel] [behaviour]".

We can look at this from the Solid QA end and make a proposal that could be used across all specs.

Reply:
I created https://github.com/solid/webid-profile/issues/101 to follow up on this.